### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by jmklix**
*Description of changes:*
standardized and restructured README.md. Moved samples into samples/README.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,35 +14,41 @@ This SDK is built on the AWS Common Runtime, a collection of libraries
...
+## Installation
+
+### Minimum Requirements
 *   Java 8 or above
 
 ** Set JAVA_HOME first
 
 *   Install Maven
-
+```
```
Comment (bretambrose) [inline README.md:35]: This is mac-specific.  I think we should remove it and if someone needs help installing maven they can use the official documentation.

File: README.md [hunk 2]
```diff
@@ -17,35 +14,41 @@ This SDK is built on the AWS Common Runtime, a collection of libraries
...
+### Build IoT Device SDK from source
 ```
 git clone https://github.com/awslabs/aws-iot-device-sdk-java-v2.git
 # update the version of the CRT being used
 mvn versions:use-latest-versions -Dincludes="software.amazon.awssdk.crt*"
 mvn clean install
 ```
 
-## Build CRT from source
+### Build CRT from source
```
Comment (bretambrose) [inline README.md:51]: I'd like Justin's feedback but I think this should be removed in favor of a link to the CRT readme.

File: README.md [hunk 3]
```diff
@@ -17,35 +14,41 @@ This SDK is built on the AWS Common Runtime, a collection of libraries
...
+### Requirements to build the AWS CRT locally
 *   CMake 3.1+
 *   Clang 3.9+ or GCC 4.4+ or MSVC 2015+
 
-## Build IoT Device SDK from source
+### Build IoT Device SDK from source
 ```
 git clone https://github.com/awslabs/aws-iot-device-sdk-java-v2.git
 # update the version of the CRT being used
 mvn versions:use-latest-versions -Dincludes="software.amazon.awssdk.crt*"
```
Comment (bretambrose) [inline README.md:47]: This seems unsafe (need Justin's feedback here).

File: README.md (hunk 2 above)
Comment (bretambrose) [inline README.md:51]: We should maybe move this section to just after the requirements and mark it optional.

File: samples/README.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "iot:Publish"
+      ],
+      "Resource": [
+        "arn:aws:iot:<b>region</b>:<b>account</b>:topic/$aws/certificates/create/json",
+        "arn:aws:iot:<b>region</b>:<b>account</b>:topic/$aws/certificates/create-from-csr/json",
+        "arn:aws:iot:<b>region</b>:<b>account</b>:topic/$aws/provisioning-templates/<b>templatename<b>/provision/json"
```
Comment (bretambrose) [inline samples/README.md:208]: As Mike pointed out in the python PR, we'll need to fix the closing tag on templatename throughout this policy.

Comment (jmklix) [reply samples/README.md]: fixed

File: README.md (hunk 1 above)
Comment (jmklix) [reply README.md]: removed

**Pull request merged**